mne/viz/_brain/_brain.py

```diff
@@ -2496,6 +2536,157 @@ def add_foci(self, coords, coords_as_verts=False, map_surface=None,
+        inst : mne.Info | mne.io.Raw | mne.Epochs | mne.Evoked
+            The MNE object with sensor locations.
+        %(trans_not_none)s
+        %(picks_all)s
+        colors : str | dict
+            The color of the sensors. If dict, channel names or types can be
+            keys with values of matplotlib colors. Default is all sensors
+            yellow and the MEG helmet and reference blue. The MEG helmet
+            can be colored with the key "meg_helmet".
+        scales : float | dict
+            The size of the sensors. If dict, channel names or types can be
+            keys with values for the size in mm. If None, default scales will
+            be used.
+        alphas : float | dict
+            The opacity of the sensors or, if dict, each sensor by name or
+            type. Default is all sensors 0.5 opacity. The MEG helmet
+            can be colored with the key "meg_helmet".
+        meg_helmet : bool
+            Whether to plot the MEG helmet (only applies if MEG sensors are
+            present in ``info``).
+        project_eeg : bool
+            Whether to project the EEG sensors to the scalp (only applies if
+            EEG sensors are present in ``info``).
+        fnirs_pairs : str | bool
+            The color to plot fnirs pairs. If False, fnirs pairs will not
+            be plotted. If None, the default color will be used.
+        fnirs_sources : str | bool
+            The color to plot fnirs sources. If False, fnirs sources will not
+            be plotted. If None, the default color will be used.
+        fnirs_detectors : str | bool
+            The color to plot the fnirs detectors. If False, fnirs detectors
+            will not be plotted. If None, the default color will be used.
+        %(verbose)s
+
+        Notes
+        -----
+        .. versionadded:: 0.24
+        """
+        from matplotlib.colors import colorConverter
+        if isinstance(inst, Info):
+            info = inst.copy()
+        else:
+            info = inst.info.copy()
+        picks = _picks_to_idx(info, picks, none='all', exclude=(),
+                              allow_empty=False)
+        info = info.pick_channels([inst.ch_names[idx] for idx in picks])
+        if trans is None:
+            warn('No `trans` provided, assuming identity head->mri transform')
+            trans = Transform('head', 'mri')
+        head_mri_t = _get_trans(trans, 'head', 'mri')[0]
+        # get transforms to "mri"window
+        to_cf_t = _get_transforms_to_coord_frame(
+            info, head_mri_t, coord_frame='mri', verbose=verbose)
+        try:
+            ch_pos, sources, detectors = _ch_pos_in_coord_frame(
+                info, to_cf_t, verbose=verbose)
+        except RuntimeError as err:
+            raise RuntimeError(
+                f'{err} You may want to use `picks=("meg", "eeg")` '
+                'to avoid sensors that should not be plotted for instance')
+        eeg_indices = pick_types(info, eeg=True)
```

Comment:
Don't exclude bads from the subset already picked
```suggestion
        eeg_indices = pick_types(info, eeg=True, exclude=())
```